src/integration-test/java/com/mongodb/jdbc/integration/testharness/TestGenerator.java

```diff
@@ -0,0 +1,136 @@
+        List<Boolean> expectedIsCaseSensitive = new ArrayList<>();
+        List<Boolean> expectedIsCurrency = new ArrayList<>();
+        List<Boolean> expectedIsDefinitelyWritable = new ArrayList<>();
+        List<Boolean> expectedIsReadOnly = new ArrayList<>();
+        List<Boolean> expectedIsSearchable = new ArrayList<>();
+        List<Boolean> expectedIsSigned = new ArrayList<>();
+        List<Boolean> expectedIsWritable = new ArrayList<>();
+
+        File directory = new File(GENERATED_TEST_DIR);
+        ResultSetMetaData resultSetMetadata = rs.getMetaData();
+        Map<String, Object> tests = new LinkedHashMap<String, Object>();
+        List<Map<String, Object>> testCases = new ArrayList<>();
+        Map<String, Object> testCase = new LinkedHashMap<String, Object>();
+        testCases.add(testCase);
+        tests.put("tests", testCases);
+
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        String fileName =
+                new SimpleDateFormat("'" + description + "'MMddHHmmss'.yaml'").format(new Date());
+
+        // generating 'expected_result'
+        ArrayList<List<Object>> result = new ArrayList<>();
+        while (rs.next()) {
+            List<Object> row = new ArrayList<>();
+            for (int i = 1; i <= resultSetMetadata.getColumnCount(); i++) {
+                row.add(rs.getObject(i));
+            }
+            result.add(row);
+        }
+
+        // generating expected resultset metadata
+        for (int i = 1; i <= resultSetMetadata.getColumnCount(); i++) {
+            expectedSqlType.add(TestTypeInfo.typesIntToString(resultSetMetadata.getColumnType(i)));
+            expectedCatalogName.add(resultSetMetadata.getCatalogName(i));
+            expectedColumnClassName.add(resultSetMetadata.getColumnClassName(i));
+            expectedColumnLabel.add(resultSetMetadata.getColumnLabel(i));
+            expectedColumnDisplaySize.add(resultSetMetadata.getColumnDisplaySize(i));
+            expectedPrecision.add(resultSetMetadata.getPrecision(i));
+            expectedScale.add(resultSetMetadata.getScale(i));
+            expectedSchemaName.add(resultSetMetadata.getSchemaName(i));
+            expectedIsAutoIncrement.add(resultSetMetadata.isAutoIncrement(i));
+            expectedIsCaseSensitive.add(resultSetMetadata.isCaseSensitive(i));
+            expectedIsCurrency.add(resultSetMetadata.isCurrency(i));
+            expectedIsDefinitelyWritable.add(resultSetMetadata.isDefinitelyWritable(i));
+            expectedIsNullable.add(
+                    TestTypeInfo.nullableIntToString(resultSetMetadata.isNullable(i)));
+            expectedIsReadOnly.add(resultSetMetadata.isReadOnly(i));
+            expectedIsSearchable.add(resultSetMetadata.isSearchable(i));
+            expectedIsSigned.add(resultSetMetadata.isSigned(i));
+            expectedIsWritable.add(resultSetMetadata.isWritable(i));
+        }
+
+        testCase.put("description", description);
```

Comment:
Should we also write the `sql` or `meta_function` that is the source of this test? And should we also include the `row_count`? And perhaps flags for whether or not the tests should be generated with `ordered` and/or `row_count_gte` set to true or false?

Reply:
Good catch, I added them to the generated output.